var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -86,15 +87,16 @@ class Hdf5(CMakePackage):
             values=('default', 'v114', 'v112', 'v110', 'v18', 'v16'),
             multi=False)
 
-    depends_on('cmake@3.12:', type='build')
-
-    depends_on('mpi', when='+mpi')
+    if sys.platform != 'win32':
+        depends_on('mpi', when='+mpi')
     depends_on('java', type=('build', 'run'), when='+java')
-    depends_on('szip', when='+szip')
+    # numactl does not currently build on darwin
+    if sys.platform != 'darwin' and sys.platform != 'win32':
+        depends_on('numactl', when='+mpi+fortran')
```

Comment:
Have you noticed that `hdf5` on Linux requires `numactl`? I just want to make sure since this PR is generally oriented at building `hdf5` on Windows, but if you noticed an additional constraint for `hdf5` on Linux then that could make sense to add here to (although it might be ideal to split that off into another PR).